pallets/validators-manager/src/lib.rs

```diff
@@ -703,10 +700,29 @@ impl<T: Config> Pallet<T> {
             <system::Pallet<T>>::block_number() >= vote.end_of_voting_period
     }
 
+    fn decompress_eth_public_key(
+        compressed_eth_public_key: ecdsa::Public,
+    ) -> Result<H512, secp256k1::Error> {
+        let decompressed = secp256k1::PublicKey::parse_slice(
+            &compressed_eth_public_key.0,
+            Some(secp256k1::PublicKeyFormat::Compressed),
+        );
+        match decompressed {
+            Ok(public_key) => {
+                let decompressed = public_key.serialize();
+                let mut m = [0u8; 64];
+                m.copy_from_slice(&decompressed[1..65]);
+                Ok(H512::from_slice(&m))
+            },
+            Err(err) => Err(err),
+        }
+    }
```

Comment:
I think this should be moved to the avn-common crate under primitives and unit-tested there.